core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -79,11 +80,15 @@ private class ClientActor(driverArgs: ClientArguments, conf: SparkConf)
           driverArgs.supervise,
           command)
 
+      for (masterActor <- mastersActor) {
         masterActor ! RequestSubmitDriver(driverDescription)
+      }
```

Comment:
Never mind, it seems that we guard against this in [this line](https://github.com/apache/spark/blob/251698fb7335a3bb465f1cd0c29e7e74e0361f4a/core/src/main/scala/org/apache/spark/deploy/master/Master.scala#L256) and already do something similar in client mode.